includes/Modules/Analytics/Measurement_Code_Injector.php

```diff
@@ -0,0 +1,108 @@
+ *
+ * @package   Google\Site_Kit\Modules\Analytics
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics;
+
+use Google\Site_Kit\Context;
+
+/**
+ * Injects Javascript based on the current active plugins
+ *
+ * Class Injector
+ */
+class Measurement_Code_Injector {
+	/**
+	 * A list of user's current active plugins that ShirshuClass supports
+	 *
+	 * @var array of string
+	 */
+	private $active_plugins = null;
+
+	/**
+	 * Holds a list of event configurations to be injected
+	 *
+	 * @var array
+	 */
+	private $event_configurations;
+
+	/**
+	 * Measurement_Event_Factory instance
+	 *
+	 * @var Measurement_Event_Factory
+	 */
+	private $event_factory = null;
+
+	/**
+	 * Plugin context.
+	 *
+	 * @since 1.0.0
+	 * @var Context
+	 */
+	protected $context;
+
+	/**
+	 * Injector constructor.
+	 *
+	 * @param array   $active_plugins list of supported plugins site currently has activated.
+	 * @param Context $module_context context used to check for AMP.
+	 */
+	public function __construct( $active_plugins, $module_context ) {
+		$this->active_plugins       = $active_plugins;
+		$this->context              = $module_context;
+		$this->event_factory        = Measurement_Event_Factory::get_instance();
+		$this->event_configurations = $this->build_event_configurations();
+		add_action( 'wp_enqueue_scripts', array( $this, 'inject_event_tracking' ), 1 );
+	}
+
+	/**
+	 * Sets the event configurations
+	 */
+	public function build_event_configurations() {
+		$event_configurations = array();
+		foreach ( $this->active_plugins as $plugin_name ) {
+			$measurement_event_list = $this->event_factory->create_measurement_event_list( $plugin_name );
+			if ( null !== $measurement_event_list ) {
+				foreach ( $measurement_event_list->get_events() as $measurement_event ) {
+					array_push( $event_configurations, $measurement_event );
+				}
+			}
+		}
+		return $event_configurations;
+	}
+
+	/**
+	 * Gets the event configurations
+	 */
+	public function get_event_configurations() {
+		return $this->event_configurations;
+	}
+
+	/**
+	 * Creates list of measurement event configurations and javascript to inject
+	 */
+	public function inject_event_tracking() {
+		if ( ! $this->context->is_amp() ) {
+			$main_file_path = plugin_basename( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+			$main_dir_path  = strstr( $main_file_path, '/', true );
+			wp_enqueue_script( // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion
+				'shirshu_inject_event_tracking',
+				'/wp-content/plugins/' . $main_dir_path . '/assets/js/modules/analytics/advanced-tracking/measurement-event-tracking.js',
+				false,
+				null,
+				false
+			);
+			wp_localize_script(
+				'shirshu_inject_event_tracking',
+				'eventConfigurations',
+				$this->event_configurations
+			);
```

Comment:
The overall approach for how to inject this code here should be slightly modified:
* Instead of loading an external file, this JavaScript should be inlined. While loading an actual JS file would generally be appropriate, the script needed here will be loaded in all users' frontends and it is fairly small - like all other Analytics `gtag` code it should be inlined.
* This can be done using `wp_add_inline_script( 'google_gtagjs', $script_code )` (see the `Analytics::enqueue_gtag_js()` method for reference). Since the script here is bigger than the very basic inline scripts already existing, you could use [PHP heredoc](https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc) to write the script into a variable and then pass that to `wp_add_inline_script`.
* Instead of injecting when `! Context::is_amp()` (which is not enough to check, since it only makes sense to inject the code when _all_ conditions for loading Site Kit's `gtag` script are met), we should directly rely on whether Site Kit already enqueued `google_gtagjs`. We can do so by calling [`wp_script_is( 'google_gtagjs' )`](https://developer.wordpress.org/reference/functions/wp_script_is/). Only if that is the case, this logic should run.
    * Since we now no longer need access to `Context::is_amp()`, the class property could be removed entirely.
* Since the `Analytics::enqueue_gtag_js()` method is hooked into WordPress at the default priority 10, the hook here should use a later priority (e.g. 11). This will need to be changed in the `add_action` call above.

Reply:
For the string variable that holds the injected script using PHP heredoc, should I make the variable a static attribute of the class or initialize it as a local variable inside the inject_event_tracking() function?